cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -788,15 +798,45 @@ public IdeSubLogger level(IdeLogLevel level) {
 
   @Override
   public void logIdeHomeAndRootStatus() {
-
     if (this.ideRoot != null) {
-      success("IDE_ROOT is set to {}", this.ideRoot);
+      success("IDE_ROOT is set to {}", formatLocationPathForDisplay(this.ideRoot));
     }
     if (this.ideHome == null) {
       warning(getMessageNotInsideIdeProject());
     } else {
-      success("IDE_HOME is set to {}", this.ideHome);
+      success("IDE_HOME is set to {}", formatLocationPathForDisplay(this.ideHome));
+    }
+  }
+
+  /**
+   * Formats a path for GDPR compliance based on the privacy mode.
+   *
+   * @param location Path to format.
+   * @return the formatted path string.
+   */
+  public String formatLocationPathForDisplay(Path location) {
+    String locationString = location.toString();
```

Comment:
in case of privacy mode this computation is done in vain.
Simply move it to an `else` block and then you do not even need the local variable `locationString`.